code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -272,6 +272,12 @@
 	for(var/ejectee in occupants)
 		mob_exit(ejectee, silent = TRUE)
 
+	/// If the former occupants get polymorphed, mutated, chestburstered,
+	/// or otherwise replaced by another mob, that mob is no longer in .occupants
+	/// and gets deleted with the mech. We need to save them.
+	for(var/mob/buggy_ejectee in contents)
+		mob_exit(buggy_ejectee, silent = TRUE)
```

Comment:
```suggestion
	/// If the former occupants get polymorphed, mutated, chestburstered,
	/// or otherwise replaced by another mob, that mob is no longer in .occupants
	/// and gets deleted with the mech. We need to save them.
	var/list/combined_components = contents ^ occupants
	for(var/mob/buggy_ejectee in combined_components)
		mob_exit(buggy_ejectee, silent = TRUE)
```

I'm not sure if the XOR operator is the best here but here's the ref https://www.byond.com/docs/ref/#/operator/^ and i can't think of anything else.

Anyways we can just distill the code here and only do one for-loop to iterate off both lists. keep in mind that this will contain things that aren't mobs as well, but the `for()` loop will explicitly check and only grab `/mob`s from the list as a part of the builtin typecheck+typecast.

Reply:
i tested it and it works so let's boogie